### ISSUE TITLE
[ENH]: Add support for separate validation and test sets in setup function.

### ISSUE THREAD

**Issue by ajaymur91** [pycaret/pycaret#3512]
### Describe the feature you want to add to this project

I am trying to create a classification model with the following workflow. 
1. Split the data into train, validation and test data
2. Compare available models
3. Fine tune the top N models (perhaps using early stopping etc) 
4. Check performance on the validation set
5. Finalize model (including the validation data) and then compare performance on test (unseen) data.
6. **Re-finalize** (including the test data) the model with best hyperparameters for deployment

Is there a workaround to achieve Step 6 with the current setup() or finalize_model()? 
If not, would this be a good feature to add?

### Describe your proposed solution

1. Add support for separate validation and test sets in setup function. 
df = pd.read_csv('data.csv')
clf = setup(df, train_size=0.8, valid_size=0.1, test_size=0.1)

2. or allow a keyword in the finalize_model()
final = finalize_model(tuned_model, data=df)

**Comment by moezali1** [comment]
Closing as part of the 4.0 backlog cleanup — see #4177.

PyCaret 4.0 is a ground-up rewrite (OOP-only API, sklearn-native engine, Python 3.11+, sklearn ≥ 1.7) and almost all pre-4.0 issues either reference removed APIs (`setup()`, `plot_model`, `evaluate_model`, `interpret_model`, `automl`, the legacy time-series module, the Streamlit dashboard, …) or no longer reproduce on the new code path.

**If this still affects you in 4.0, please re-file as a new issue** with:

- the experiment constructor call (`Experiment(...)`),
- the operation that fails,
- the full traceback,
- your Python / sklearn / pycaret versions.

Resources for 4.0:

- Docs: https://pycaret.org/docs
- Migrate from 3.x: https://pycaret.org/docs/guides/migrating-from-3-x
- API reference: https://pycaret.org/reference
- FAQ: https://pycaret.org/docs/resources/faqs

Thanks for filing — and apologies for the long latency. The repo had drifted; we're starting clean.
